EDSC-1113, EDSC-1114: Spatial and temporal extraction

mightynimble opened the pull request.

macrouch commented:
This doesn't look good. The box also looks too narrow on the /search page. Maybe a class is missing?
![screen shot 2016-08-31 at 11 28 20 am](https://cloud.githubusercontent.com/assets/171388/18134912/122ef400-6f6e-11e6-8c75-df62d0d1b2f2.png)

macrouch commented on spec/features/address_bar_spec.rb:
```diff
@@ -476,12 +476,13 @@ def project_id
 
         before(:each) do
           before_project_id = project_id
+          p  "--------------- project_Id : #{project_id}"
```
remove debugging code

macrouch commented:
Setup steps for the new service need to be added to the readme file (or at least a line in the Prerequisites section)

macrouch commented:
There should be new tests added. I don't see any tests that use the new functionality specifically.

mightynimble commented:
I will address the comments in next commit.
I am working on the tests right now but may not be able to finish them in time. Just want you to view the code changes so that when the tests are added and the build is green, @mreese84 can approve it and merge to master in time.

macrouch commented on app/assets/stylesheets/modules/nlp_parsing.css.scss:
```diff
@@ -0,0 +1,39 @@
+// The .tt- prefix is used by the typeahead plugin.  This file
```
If we aren't using the typeahead input field anymore, we should use new class names instead of typeahead specific names (i.e "tt-")

mightynimble replied:
will do.

macrouch commented on README.md:
```diff
@@ -47,6 +47,7 @@ See public/licenses.txt
 * A Ruby manager such as [RVM](http://rvm.io/) or [rbenv](https://github.com/rbenv/rbenv) is strongly recommended.
 * (For shapefile support) access to an [ogre](http://ogre.adc4gis.com) server
 * (For placename completion) a [GeoNames](http://www.geonames.org) account
+* For automatic spatial and temporal extraction from the search text, clone and set up an [EDSC-NLP](https://github.com/mightynimble/edsc-nlp) server
```
Should this live under NASA instead of your personal GitHub? I'm not sure what we need to do for that

mightynimble replied:
I have no idea either. I will use my repo for now and will update the README when we know where to put it.

mreese84 replied:
I'll ping the higher-ups to see what they think. My gut tells me it shouldn't live in a personal account too, but I'm not sure where to put it either. I'll take that action item.

bilts replied:
We have the ability to request additional repos from the NASA GitHub org.

On Aug 31, 2016 10:35 PM, "Mark Reese" <notifications@github.com> wrote:

> In README.md
> <https://github.com/nasa/earthdata-search/pull/251#discussion_r77106368>:
>
> > @@ -47,6 +47,7 @@ See public/licenses.txt
> >  * A Ruby manager such as [RVM](http://rvm.io/) or [rbenv](https://github.com/rbenv/rbenv) is strongly recommended.
> >  * (For shapefile support) access to an [ogre](http://ogre.adc4gis.com) server
> >  * (For placename completion) a [GeoNames](http://www.geonames.org) account
> > +* For automatic spatial and temporal extraction from the search text, clone and set up an [EDSC-NLP](https://github.com/mightynimble/edsc-nlp) server
>
> I'll ping the higher-ups to see what they think. My gut tells me it
> shouldn't live in a personal account too, but I'm not sure where to put it
> either. I'll take that action item.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/nasa/earthdata-search/pull/251/files/f068791d4d903c42face4463846f7777b7ce8670..e605935d160b6a127a765d39ae9de0f02cf71c57#r77106368>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AADuwMU0b2p6Z6470GlFCsjhazKJmlURks5qljnygaJpZM4JxrgI>
> .
>

macrouch commented on spec/features/map/nlp_spatial_temporal_extraction_spec.rb:
```diff
@@ -0,0 +1,36 @@
+require "spec_helper"
+
+describe "Spatial and temporal extraction", reset: false do
```
This is a good start. Do I remember talk of being able to pull both spatial and temporal out of the same search? If so be sure to add that test as well

mightynimble replied:
Yup, both spatial and temporal will be extracted from the search text.

mightynimble commented:
@mreese84 I have asked Matthew to review the code changes and addressed all the comments. Since he is out tomorrow. Could you take over the review? I added a couple of new tests and made a css change in the last two commits.

mreese84 commented:
Yeah. I'm getting the NLP stuff setup now

On Wednesday, August 31, 2016, Bo Sun <notifications@github.com> wrote:

> @mreese84 <https://github.com/mreese84> I have asked Matthew to review
> the code changes and addressed all the comments. Since he is out tomorrow.
> Could you take over the review? I added a couple of new tests and made a
> css change in the last two commits.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/nasa/earthdata-search/pull/251#issuecomment-243901718>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHUwd59gzoHABa4mqRd2jnJrvWdyvKxyks5qlezFgaJpZM4JxrgI>
> .
>


-- 
*_________________*
*Mark Reese*
Element 84 | NASA Earthdata
972.816.1653

mreese84 commented:
@mightynimble When searching for "dallas" the spatial extraction doesn't appear to work. However, if I search for "Dallas" it does work. We shouldn't force users to capitalize search terms, even place names.

mreese84 commented:
@mightynimble if I manually clear out the spatial coordinates I would expect the search to be updated so that more collections are returned. Currently nothing happens when I manually clear those coordinates. This should be have similar to clearing the temporal range (the query is updated and searched against.)

mreese84 commented:
@mightynimble please update the destination for this merge to the ui-sandbox branch.

mightynimble commented:
@mreese84 the spatial is not cleared out when the coordinates are cleared also happens in SIT/UAT/PROD. This is a separate issue and I couldn't resolve it in a short time. I'd say file a new ticket and circle back to it later.

mightynimble commented:
Dana agreed to merge this to master.
